src/object_based/extent_htm.cpp

```diff
@@ -0,0 +1,352 @@
+#include "extent_htm.hpp"
+#include <Eigen/Core>
+#include <memory>
+
+#include "../common/geom.hpp"
+#include "../common/helpers/eigen_helpers.hpp"
+
+const double PI = boost::math::constants::pi<double>();
+
+namespace {
+  bool pointInsideSphericalTriangle(
+      const Eigen::Ref<const Eigen::Vector3d> &p1,
+      const Eigen::Ref<const Eigen::Vector3d> &p2,
+      const Eigen::Ref<const Eigen::Vector3d> &p3,
+      const Eigen::Ref<const Eigen::Vector3d> &q) {
+    return p2.cross(p1).dot(q) >= -1e-6 && p3.cross(p2).dot(q) >= -1e-6 &&
+           p1.cross(p3).dot(q) >= -1e-6;
+  }
+}  // namespace
+
+namespace ear {
+  WeightingFunctionEdges::WeightingFunctionEdges(const WeightingFunction &wf)
+      : wf(wf) {
+    if (wf.is_circular) {
```

Comment:
If 'inside_edges' / 'outside_edges' aren't touched after construction and this gets used as part of the panning algo, might be worth a call to `reserve()` before adding elements in both the if/else cases so the vectors are as small as possible / might end up cached more often?